File: app/src/main/java/com/sp/base/network/ApiService.java
package com.sp.base.network;


import com.google.gson.JsonElement;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;

import static com.sp.base.network.NetworkingConstants.GETPOCKEMON;

public interface ApiService {

//    @POST()
//    Observable<JsonElement> GetRouteList(@Url String url, @Body GetRouteListRequest param);

    @GET(GETPOCKEMON)
    Observable<JsonElement> getDataList();
}

File: app/src/main/java/com/baseapplication/network/ApiService.kt
package com.baseapplication.network

import com.google.gson.JsonElement
import com.baseapplication.network.NetworkingConstants.Companion.DUMMY
import com.baseapplication.network.NetworkingConstants.Companion.GETPOCKEMON
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Query

open interface ApiService {
    @get:GET(GETPOCKEMON)
    val dataList: Observable<JsonElement>?

    @GET(DUMMY)
    fun getDummyLIst(
        @Query("page") page: Int,
        @Query("limit") lmit: Int
    ): Observable<JsonElement>? //page=1&limit=10"

     @GET(DUMMY)
    suspend fun getDataList(
        @Query("page") page: Int,
        @Query("limit") lmit: Int
    ): JsonElement? //page=1&limit=10"
}